entity-framework/core/providers/cosmos/modeling.md

```diff
@@ -0,0 +1,334 @@
+---
+title: Modeling - Azure Cosmos DB Provider - EF Core
+description: Configuring the model with the Azure Cosmos DB EF Core Provider
+author: roji
+ms.date: 09/19/2024
+uid: core/providers/cosmos/modeling
+---
+# Configuring the model with the EF Core Azure Cosmos DB Provider
+
+## Containers and entity types
+
+In Azure Cosmos DB, JSON documents are stored in containers. Unlike tables in relational databases, Cosmos DB containers can contain documents with different shapes - a container does not impose a uniform schema on its documents. However, various configuration options are defined at the container level, and therefore affect all documents contained within it. See the [Cosmos DB documentation on containers](/azure/cosmos-db/resource-model) for more information.
+
+By default, EF maps all entity types to the same container; this is usually a good default in terms of performance and pricing. The default container is named after the your .NET context type (`OrderContext` in this case). To change the default container name, use <xref:Microsoft.EntityFrameworkCore.CosmosModelBuilderExtensions.HasDefaultContainer%2A>:
+
+```csharp
+modelBuilder.HasDefaultContainer("Store");
+```
+
+To map an entity type to a different container use <xref:Microsoft.EntityFrameworkCore.CosmosEntityTypeBuilderExtensions.ToContainer%2A>:
+
+```csharp
+modelBuilder.Entity<Order>().ToContainer("Orders");
+```
+
+Before mapping entity types to different containers, make sure you understand the potential performance and pricing implications (e.g. with regards to dedicated and shared throughput); [see the Cosmos DB documentation to learn more](/azure/cosmos-db/resource-model).
+
+## IDs and keys
+
+Cosmos DB requires all documents to have an `id` JSON property which uniquely identifies them. Like other EF providers, the EF Cosmos provider will attempt to find a property named `Id` or `<type name>Id`, and configure that property as the key of your entity type, mapping it to the `id` JSON property. You can configure any property to be the key property by using <xref:Microsoft.EntityFrameworkCore.Metadata.Builders.EntityTypeBuilder%601.HasKey%2A>; see [the general EF documentation on keys](xref:core/modeling/keys) for more information.
+
+Developers coming to Cosmos DB from other database sometimes expect the key (`Id`) property to be generated automatically. For example, on SQL Server, EF configures numeric key properties to be IDENTITY columns, where auto-incrementing values are generated in the database. In contrast, Cosmos DB does not support automatic generation of properties, and so key properties must be explicitly set. Inserting an entity type with an unset key property will simply insert the CLR default value for that property (e.g. 0 for `int`), and a second insert will fail; EF issues a warning if you attempt to do this.
+
+If you'd like to have a GUID as your key property, you can configure EF to generate random values at the client:
```

Comment:
```suggestion
If you'd like to have a GUID as your key property, you can configure EF to generate unique values at the client:
```

Reply:
Changing to "unique, random values" to make it clear that it's random (and e.g. not some sort of sequential thing)